Source/DafnyCore/Dafny.atg

```diff
@@ -3952,15 +3957,20 @@ LogicalExpression<out Expression e0, bool allowLemma, bool allowLambda, bool all
       OrOp                                                      (. x = t; .)
       RelationalExpression<out e1, allowLemma, allowLambda, allowBitwiseOps>
          (. startToken = e0.StartToken;
-            e0 = new BinaryExpr(x, BinaryExpr.Opcode.Or, e0, e1, firstOp);
-            e0.RangeToken = new RangeToken(startToken, t); .)    }
+            var fmtToken = e0.RangeToken.EndToken.Next;
+            e0 = new BinaryExpr(x, BinaryExpr.Opcode.Or, e0, e1);
+            e0.RangeToken = new RangeToken(startToken, t);
+            e0.FormatTokens = new[] { fmtToken };
+         .) }
     [ IF(IsAndOp()) AndOp (. SemErr(t, "Ambiguous use of && and ||. Use parentheses to disambiguate."); .) ]
     (. e0.RangeToken = new RangeToken(firstOp, t);
        if (e0 == first) {
          // There was only one disjunct. To make sure that the type checker still checks it to
          // be a boolean, we disjoin [sic] "false" to its left.
-         e0 = new BinaryExpr(x, BinaryExpr.Opcode.Or, new LiteralExpr(new AutoGeneratedToken(x), false), e0, firstOp);
+         var fmtToken = e0.RangeToken.EndToken.Next;
+         e0 = new BinaryExpr(x, BinaryExpr.Opcode.Or, new LiteralExpr(new AutoGeneratedToken(x), false), e0);
          e0.RangeToken = new RangeToken(firstOp, t);
+         e0.FormatTokens = new[] { fmtToken };
```

Comment:
That seems wrong. If we only have

`&& A`
then e0  is A and e0.RangeToken.EndToken is the token after A. This is not what we want, right?

Reply:
The use of `RangeToken.EndToken.Next` was originally a suggestion from @keyboardDrummer, and it seems to have the correct behavior, even though it does surprise me. I'm curious whether he can say anything about the reasoning he had.

Reply:
It was just a guess. What does it mean to have only `&& A` @MikaelMayer ?